File: front/src/app/Models/SpermAnalysis.ts
import { SpermNorm } from "./SpermNorm";

export class SpermAnalysis {
  id!: number;
  date!: Date;
  abstinence!: number;
  ph!: number;
  volume!: number;
  concentration!: number;
  progressivemobility!: number;
  totalmotility!: number;
  totalcount!: number;
  roundcells!: number;
  leukocytes!: number;
  morphology!: number;
  norm!: typeof SpermNorm[keyof typeof SpermNorm];
  vitality!: number;
  tms!: number;
  consultationId!: number;
}

File: Services/MedicalReport/src/main/java/org/example/medicalreport/Models/AnalyseDTOs/SpermAnalysis.java
package org.example.medicalreport.Models.AnalyseDTOs;

import lombok.Data;

import java.time.LocalDate;

@Data
public class SpermAnalysis {
    long id;
    LocalDate date;
    int abstinence;
    double ph;
    double volume;
    double concentration;
    double progressivemobility;
    double totalmotility;
    double totalcount;
    double roundcells;
    double leukocytes;
    double morphology;
    int norm;
    double vitality;
    double tms;
    long consultationId;
}
